Index: gtk/gtkmodules.c
===================================================================
--- gtk/gtkmodules.c	(revision 19134)
+++ gtk/gtkmodules.c	(working copy)
@@ -32,6 +32,10 @@
 #include <gmodule.h>
 #include <pango/pango-utils.h>	/* For pango_split_file_list */
 
+#ifdef GDK_WINDOWING_QUARTZ
+#include <dlfcn.h>
+#endif
+
 typedef struct _GtkModuleInfo GtkModuleInfo;
 struct _GtkModuleInfo
 {
@@ -75,7 +79,21 @@ get_module_path (void)
   if (exe_prefix)
     default_dir = g_build_filename (exe_prefix, "lib", "gtk-2.0", NULL);
   else
+#ifdef GDK_WINDOWING_QUARTZ
+    {
+      const char * (* get_bundle_prefix) (void);
+
+      get_bundle_prefix = dlsym (RTLD_DEFAULT,
+                                 "_gtk_framework_get_bundle_prefix");
+      if (get_bundle_prefix)
+        default_dir = g_build_filename (get_bundle_prefix (), "Resources",
+                                        "lib", "gtk-2.0", NULL);
+      else
+        default_dir = g_build_filename (GTK_LIBDIR, "gtk-2.0", NULL);
+    }
+#else
     default_dir = g_build_filename (GTK_LIBDIR, "gtk-2.0", NULL);
+#endif
 
   if (module_path_env && home_gtk_dir)
     module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
Index: gtk/gtkrc.c
===================================================================
--- gtk/gtkrc.c	(revision 19134)
+++ gtk/gtkrc.c	(working copy)
@@ -62,6 +62,10 @@
 #include <io.h>
 #endif
 
+#ifdef GDK_WINDOWING_QUARTZ
+#include <dlfcn.h>
+#endif
+
 typedef struct _GtkRcSet    GtkRcSet;
 typedef struct _GtkRcNode   GtkRcNode;
 typedef struct _GtkRcFile   GtkRcFile;
@@ -527,6 +531,22 @@ gtk_rc_add_initial_default_files (void)
       gtk_rc_add_default_file (str);
       g_free (str);
 
+#ifdef GDK_WINDOWING_QUARTZ
+      {
+        const char * (* get_bundle_prefix) (void);
+
+        get_bundle_prefix = dlsym (RTLD_DEFAULT,
+                                   "_gtk_framework_get_bundle_prefix");
+        if (get_bundle_prefix)
+          {
+            str = g_build_filename (get_bundle_prefix (), "Resources",
+                                    "etc", "gtk-2.0", "gtkrc", NULL);
+            gtk_rc_add_default_file (str);
+            g_free (str);
+          }
+      }
+#endif
+
       home = g_get_home_dir ();
       if (home)
 	{
