Index: pango/pango-utils.c
===================================================================
--- pango/pango-utils.c	(revision 2502)
+++ pango/pango-utils.c	(working copy)
@@ -581,6 +585,20 @@ read_config_file (const char *filename, 
 	      gchar *value =  g_key_file_get_value(key_file, group, key, &key_error);
 	      if (value != NULL)
 		{
+		  if (!g_path_is_absolute (value))
+		    {
+		      gchar *dirname;
+		      gchar *absolute_path;
+
+		      dirname = g_path_get_dirname (filename);
+		      absolute_path = g_build_filename (dirname, value, NULL);
+
+		      g_free (dirname);
+		      g_free (value);
+
+		      value = absolute_path;
+		    }
+
 		  g_hash_table_insert (config_hash,
 				       g_strdup_printf ("%s/%s", group, key),
 				       value);
